packages/amplication-server/src/core/resource/resource.service.ts

```diff
@@ -381,6 +389,88 @@ export class ResourceService {
     return resource;
   }
 
+  async createPreviewService({
+    args,
+    user,
+    pluginsToInstall,
+    requireAuthenticationEntity,
+  }: CreatePreviewServiceArgs): Promise<Resource> {
+    const { serviceSettings, gitRepository, ...rest } = args.data;
+    const resource = await this.createResource(
+      {
+        data: {
+          ...rest,
+          resourceType: EnumResourceType.Service,
+        },
+      },
+      user,
+      gitRepository,
+      "create resource"
+    );
+
+    await this.prisma.resourceRole.create({
+      data: { ...USER_RESOURCE_ROLE, resourceId: resource.id },
+    });
+
+    if (requireAuthenticationEntity) {
+      await this.entityService.createDefaultEntities(resource.id, user);
+      serviceSettings.authEntityName = USER_ENTITY_NAME;
+    }
+
+    await this.serviceSettingsService.createDefaultServiceSettings(
+      resource.id,
+      user,
+      serviceSettings
+    );
+
+    const defaultAuthPlugins: PluginInstallationCreateInput[] = [
+      {
+        displayName: "Auth-core",
+        pluginId: "auth-core",
+        npm: "@amplication/plugin-auth-core",
+        version: "latest",
+        enabled: true,
+        resource: { connect: { id: resource.id } },
+      },
+      {
+        displayName: "Auth-jwt",
+        pluginId: "auth-jwt",
+        npm: "@amplication/plugin-auth-jwt",
+        version: "latest",
+        enabled: true,
+        resource: { connect: { id: resource.id } },
+      },
+    ];
+
+    const defaultDBPlugin: PluginInstallationCreateInput = {
+      displayName: "postgres",
+      pluginId: "db-postgres",
+      npm: "@amplication/plugin-db-postgres",
+      version: "latest",
+      enabled: true,
+      resource: { connect: { id: resource.id } },
+    };
+
+    const plugins = [
```

Comment:
So this answers another question I asked about installing the default plugins. I find it confusing that this method accepts "pluginsToInstall" as a parameter but still might install up to 3 other plugins. This is very confusing / not intuitive. Why aren't those plugins passed in from the caller as "pluginsToInstall", or remove this parameter and allow a separate way to install plugins?

Reply:
Let's just change the parameter name from `pluginsToInstall` to `nonDefaultPluginsToInstall` or `additionalPluginsToInstall` or similar, to make it clearer

Reply:
as I see it, `pluginsToInstall` (additionalPluginsToInstall) should be plugins that in the future we might want to use for other preview features. For now, it passed as a hard-coded value (empty array), but in the future, it might be a calculated value based on the preview account type.

`createPreviewService` handles the installation of the default plugins as without these plugins the service will not function as expected.